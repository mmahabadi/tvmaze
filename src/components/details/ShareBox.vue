<script setup lang="ts">
import { defineProps } from 'vue'
import ShareIcon from '@/@ui/components/icons/ShareIcon.vue'
import IconPlay from '@/@ui/components/icons/IconPlay.vue'

defineProps<{
  language: string
}>()
</script>
<template>
  <div class="box">
    <div class="col1">
      <button class="share-btn">
        <ShareIcon />
      </button>
    </div>
    <div class="col2">
      <p>
        Language : <span class="ml-2 truncate">{{ language }}</span>
      </p>
    </div>
    <div class="col3">
      <button class="play-btn">
        <IconPlay />
        Watch
      </button>
    </div>
  </div>
</template>
<style scoped>
.box {
  @apply grid sm:grid-cols-5 grid-cols-3 gap-4 p-6 items-center bg-gray-900 border border-gray-800 rounded-lg;

  .col1 {
    @apply col-span-1 border-r hidden sm:flex;
  }
  .col2 {
    @apply col-span-2 flex-col hidden sm:flex font-medium text-sm;
  }
  .col3 {
    @apply sm:col-span-2 col-span-3 flex justify-end font-medium text-sm;
  }

  .share-btn {
    @apply w-10 h-10 flex justify-center items-center rounded-lg bg-white bg-opacity-20;
  }
  .play-btn {
    @apply py-4 transition border-2 rounded-full flex justify-center items-center gap-4 w-full sm:py-3;
    border-color: var(--primary-color);
  }
  .share-btn:hover,
  .play-btn:hover {
    background-color: var(--primary-color);
  }
}
</style>
