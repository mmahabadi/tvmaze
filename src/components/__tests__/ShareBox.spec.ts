import { describe, it, expect, beforeEach } from 'vitest'
import { mount, VueWrapper } from '@vue/test-utils'
import ShareBox from '../details/ShareBox.vue'
import ShareIcon from '@/@ui/components/icons/ShareIcon.vue'
import IconPlay from '@/components/icons/IconPlay.vue'

describe('DetailsPoster', () => {
  let wrapper: VueWrapper<unknown>

  beforeEach(() => {
    wrapper = mount(ShareBox, {
      props: { language: 'English' },
      global: {
        stubs: {
          ShareIcon,
          IconPlay
        }
      }
    })
  })
  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
