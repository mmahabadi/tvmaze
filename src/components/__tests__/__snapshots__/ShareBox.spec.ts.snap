// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DetailsPoster > renders correctly 1`] = `
"<div data-v-b00b0726="" class="box">
  <div data-v-b00b0726="" class="col1"><button data-v-b00b0726="" class="share-btn"><svg data-v-b00b0726="" stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 448 512" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg">
        <path d="M352 320c-22.608 0-43.387 7.819-59.79 20.895l-102.486-64.054a96.551 96.551 0 0 0 0-41.683l102.486-64.054C308.613 184.181 329.392 192 352 192c53.019 0 96-42.981 96-96S405.019 0 352 0s-96 42.981-96 96c0 7.158.79 14.13 2.276 20.841L155.79 180.895C139.387 167.819 118.608 160 96 160c-53.019 0-96 42.981-96 96s42.981 96 96 96c22.608 0 43.387-7.819 59.79-20.895l102.486 64.054A96.301 96.301 0 0 0 256 416c0 53.019 42.981 96 96 96s96-42.981 96-96-42.981-96-96-96z"></path>
      </svg></button></div>
  <div data-v-b00b0726="" class="col2">
    <p data-v-b00b0726=""> Language : <span data-v-b00b0726="" class="ml-2 truncate">English</span></p>
  </div>
  <div data-v-b00b0726="" class="col3"><button data-v-b00b0726="" class="play-btn"><svg data-v-b00b0726="" class="svg-icon" style="width: 1em; height: 1em; vertical-align: middle; fill: currentColor; overflow: hidden;" viewBox="0 0 1024 1024" version="1.1" xmlns="http://www.w3.org/2000/svg">
        <path d="M852.727563 392.447107C956.997809 458.473635 956.941389 565.559517 852.727563 631.55032L281.888889 993.019655C177.618644 1059.046186 93.090909 1016.054114 93.090909 897.137364L93.090909 126.860063C93.090909 7.879206 177.675064-35.013033 281.888889 30.977769L852.727563 392.447107 852.727563 392.447107Z"></path>
      </svg> Watch </button></div>
</div>"
`;
